mirrord-layer/src/config.rs

```diff
@@ -1,66 +1,26 @@
-use envconfig::Envconfig;
+pub mod env;
+pub mod file;
+pub mod util;
 
-#[derive(Envconfig, Debug, Clone)]
+#[derive(Debug, Clone)]
```

Comment:
All these config structs could be dynamically generated with like a third of the code if Rust let you do that sort of thing, but since it doesn't, what if they were HashMaps themselves? So you would define the schema in a HashMap, then generate HashMaps for instances of the configuration at runtime? Or something similar but more Rusty, if it exists.